engine/engine.go

```diff
@@ -5,44 +5,33 @@ import (
 	"errors"
 	"sync"
 
+	"github.com/filecoin-project/go-legs"
+	"github.com/filecoin-project/indexer-reference-provider"
 	"github.com/filecoin-project/indexer-reference-provider/config"
-	"github.com/filecoin-project/indexer-reference-provider/core"
-	icl "github.com/filecoin-project/storetheindex/api/v0/ingest/client/libp2p"
-	schema "github.com/filecoin-project/storetheindex/api/v0/ingest/schema"
+	"github.com/filecoin-project/storetheindex/api/v0/ingest/schema"
 	"github.com/ipfs/go-cid"
 	"github.com/ipfs/go-datastore"
 	dssync "github.com/ipfs/go-datastore/sync"
 	logging "github.com/ipfs/go-log/v2"
 	"github.com/ipld/go-ipld-prime"
-
 	cidlink "github.com/ipld/go-ipld-prime/linking/cid"
-	crypto "github.com/libp2p/go-libp2p-core/crypto"
+	"github.com/libp2p/go-libp2p-core/crypto"
 	"github.com/libp2p/go-libp2p-core/host"
-	"github.com/libp2p/go-libp2p-core/peer"
-	"github.com/multiformats/go-multihash"
-
-	legs "github.com/filecoin-project/go-legs"
 )
 
 var log = logging.Logger("provider/engine")
 
-var _ core.Interface = &Engine{}
+var _ provider.Interface = (*Engine)(nil)
 
-// NOTE: Consider including this constant as config or options
+// TODO: Consider including this constant as config or options
 // in provider so they are conigurable.
-const (
-	// metadataProtocol identifies the protocol used by provider
-	// to encode metadata.
-	metadataProtocol = 0
-	// MaxCidsInChunk number of entries to include in each chunk
-	// of ingestion linked list.
-	MaxCidsInChunk = 100
-)
 
-var (
-	// ErrNoCallback is thrown when no callback has been defined.
-	ErrNoCallback = errors.New("no callback was registered in indexer")
-)
+// MaxCidsInChunk number of entries to include in each chunk
```

Comment:
```suggestion
// MaxCidsInChunk is the maximum number of entries to include in each chunk
```

Reply:
I ended up unexporting this and dropped `Cid` in name.